src/test/java/com/aws/greengrass/clientdevices/auth/metrics/MetricsTest.java

```diff
@@ -93,14 +98,70 @@ void GIVEN_certificateSubscriptionEvents_WHEN_eventsEmitted_THEN_subscribeFailur
+        // Emitting multiple Verify Client Device Identity events to ensure the metric is incremented correctly
+        domainEvents.emit(new VerifyClientDeviceIdentityEvent(VerifyClientDeviceIdentityEvent.VerificationStatus.SUCCESS));
+        domainEvents.emit(new VerifyClientDeviceIdentityEvent(VerifyClientDeviceIdentityEvent.VerificationStatus.SUCCESS));
+        domainEvents.emit(new VerifyClientDeviceIdentityEvent(VerifyClientDeviceIdentityEvent.VerificationStatus.SUCCESS));
+
+        // Checking that the emitter collects the metrics as expected
+        Metric metric = Metric.builder()
+                .namespace("aws.greengrass.clientdevices.Auth")
+                .name(ClientDeviceAuthMetrics.METRIC_VERIFY_CLIENT_DEVICE_IDENTITY_SUCCESS)
+                .unit(TelemetryUnit.Count)
+                .aggregation(TelemetryAggregation.Sum)
+                .value(3L)
+                .timestamp(Instant.now(clock).toEpochMilli())
+                .build();
+
+        List<Metric> collectedMetrics = metrics.collectMetrics();
+        Metric verifyDeviceIdentitySuccess = collectedMetrics.get(2);
+
+        assertEquals(metric.getValue(), verifyDeviceIdentitySuccess.getValue());
+        assertEquals(metric.getName(), verifyDeviceIdentitySuccess.getName());
+        assertEquals(metric.getTimestamp(), verifyDeviceIdentitySuccess.getTimestamp());
+        assertEquals(metric.getAggregation(), verifyDeviceIdentitySuccess.getAggregation());
+        assertEquals(metric.getUnit(), verifyDeviceIdentitySuccess.getUnit());
+        assertEquals(metric.getNamespace(), verifyDeviceIdentitySuccess.getNamespace());
+    }
+
+    @Test
+    void GIVEN_verifyClientDeviceIdentityEvents_WHEN_eventsEmitted_THEN_verifyDeviceIdentityFailMetricCorrectlyEmitted() {
+        // Emitting multiple Verify Client Device Identity events to ensure the metric is incremented correctly
+        domainEvents.emit(new VerifyClientDeviceIdentityEvent(VerifyClientDeviceIdentityEvent.VerificationStatus.FAIL));
+        domainEvents.emit(new VerifyClientDeviceIdentityEvent(VerifyClientDeviceIdentityEvent.VerificationStatus.FAIL));
+        domainEvents.emit(new VerifyClientDeviceIdentityEvent(VerifyClientDeviceIdentityEvent.VerificationStatus.FAIL));
+
+        // Checking that the emitter collects the metrics as expected
+        Metric metric = Metric.builder()
+                .namespace("aws.greengrass.clientdevices.Auth")
+                .name(ClientDeviceAuthMetrics.METRIC_VERIFY_CLIENT_DEVICE_IDENTITY_FAILURE)
+                .unit(TelemetryUnit.Count)
+                .aggregation(TelemetryAggregation.Sum)
+                .value(3L)
+                .timestamp(Instant.now(clock).toEpochMilli())
+                .build();
+
+        List<Metric> collectedMetrics = metrics.collectMetrics();
+        Metric verifyDeviceIdentityFail = collectedMetrics.get(3);
```

Comment:
index is pretty fragile, can we search for the metric by name?  same question for other tests that do this